conn.go

```diff
@@ -495,6 +518,16 @@ func (conn *Conn) sendReply(dest string, serial uint32, values ...interface{}) {
 	conn.outLck.RUnlock()
 }
 
+func (conn *Conn) defaultSignalAction(fn func(h *defaultSignalHandler, ch chan<- *Signal), ch chan<- *Signal) {
+	conn.signalLck.RLock()
+	defer conn.signalLck.RUnlock()
+	if !isDefaultSignalHandler(conn.signalHandler) {
```

Comment:
Why not just have AddSignal and RemoveSignal part of the SignalHandler interface so that any registered signal handler can work with the existing API? It doesn't prevent the registered signal handler from having other APIs, for different schemes of signal handling, but it means we remove these checks for the common case of the default being used.